lib/Deprecated.js

```diff
@@ -676,26 +676,27 @@ class iDataQueue {
         cb(null);
         return;
       }
-      parseString(str, (parseError, result) => {
-        if (parseError) {
-          if (this.reportError) {
-            cb(parseError, null);
-            return;
-          }
-          cb(str);
-          return;
-        }
-        if (result.myscript.pgm[0].success && result.myscript.pgm[0].success[0].includes('+++ success')) {
-          rtValue = true;
-        } else {
-          rtValue = str;
-        }
+      const parser = new XMLParser();
+      const result = parser.parse(str);
+      if (Object.keys(result).length === 0) {
```

Comment:
@sravani1510 

I took a look at the fast xml parser docs and found out about `XMLValidator`

https://github.com/NaturalIntelligence/fast-xml-parser/blob/master/docs/v4/4.XMLValidator.md

I think we should be validating the XML with XMLValidator instead of checking the object keys

Reply:
We should probably do the same for the checks in the tests too https://github.com/IBM/nodejs-itoolkit/pull/400